packages/devtools_app/lib/src/timeline/timeline_controller.dart

```diff
@@ -245,7 +245,9 @@ class TimelineController implements DisposableController {
 
   Future<void> _offlineModeChanged() async {
     await clearData();
-    await timelineService.updateListeningState(true);
+    if (serviceManager.connectedApp != null) {
```

Comment:
`updateListeningState` awaits `serviceManager.serviceAvailable.future;` IIRC there can be a race between the html app calling updateListeningState and the service becoming available, so we await so that the listening state is updated once the service connects.

If we add the `serviceManager.connectedApp != null` check there, then we can't await until the service is available.

I'm hesitant to do this and break the html app.

Reply:
I will add a TODO to re-evaluate once we delete the html app.

Reply:
sgtm